src/explorer/schemaTreeItem.ts

```diff
@@ -97,31 +97,27 @@ export default class SchemaTreeItem extends vscode.TreeItem
       return Promise.resolve([]);
     }
 
-    if (this.childrenCacheIsUpToDate) {
+    if (this.cacheIsUpToDate) {
+      const pastChildrenCache = this.childrenCache;
+      this.childrenCache = {};
+
+      // We manually rebuild each node to ensure we update the expanded state.
+      Object.keys(pastChildrenCache).forEach(fieldName => {
+        this.childrenCache[fieldName] = new FieldTreeItem(
+          pastChildrenCache[fieldName].field,
+          pastChildrenCache[fieldName].isExpanded,
+          pastChildrenCache[fieldName].getChildrenCache()
+        );
+      });
+
       if (!this.hasClickedShowMoreFields && this.hasMoreFieldsToShow) {
         return Promise.resolve([
-          ...Object.values(this.childrenCache).map(
-            (cachedField) =>
-              new FieldTreeItem(
-                cachedField.field,
-                cachedField.isExpanded,
-                cachedField.getChildrenCache()
-              )
-          ),
```

Comment:
Here was the bug with the schema. We were rebuilding the schema's tree items in order to control their expanded state, however we weren't updating the cache with the newly created items. Because of that we were accessing some old nodes.